File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
               int res = 0;
       for(int i=0;i< s.size();i++){
            res = res+ check(s, i, i);
         
       } 
        for(int i=0;i< s.size()-1;i++){
            res  = res+check(s, i, i+1);
            
       } 
       return res;
     
    }
    int  check(string& s, int l, int r){
        int ans = 0;
        while(l >=0 && r < s.size() && s[l] == s[r]){
            l = l-1;
            r = r+1;
            ans++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/max-area-of-island/submission-0.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        seen = set()
        M = len(grid)
        N = len(grid[0])
        res = 0
        ans = 0
        directions = [[0,1],[1,0],[-1,0],[0,-1]]

        def dfs(i,j):
            nonlocal ans
            nonlocal res
            res +=1
            ans = max(res, ans)
            # print("in dfs and i, j are : ", i , j)
            for x,y in directions:
                r = i + x
                c = j + y
                if r >=0 and c >=0 and r < M and c < N and grid[r][c] == 1 and  (r,c) not in seen:
                    seen.add((r,c))
                    dfs(r,c)    
        for i in range(M):
            for j in range(N):
                if grid[i][j] == 1 and (i,j) not in seen:
                    seen.add((i,j))
                    res = 0
                    # print("DFS CALLED AT : ", i, j)
                    dfs(i,j)
        return ans